staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -61,6 +70,18 @@ type CustomResourceDefinitionVersion struct {
 	// Storage flags the version as storage version. There must be exactly one flagged
 	// as storage version.
 	Storage bool
+	// Schema describes the schema for CustomResource used in validation, pruning, and defaulting.
+	// Setting this value overrides the global validation schema for this version.
+	// +optional
+	Schema *JSONSchemaProps
+	// Subresources describes the subresources for CustomResources
+	// Setting this value overrides the global subresources for this version.
+	// +optional
+	Subresources *CustomResourceSubresources
+	// AdditionalPrinterColumns are additional columns shown e.g. in kubectl next to the name. Defaults to a created-at column.
+	// Setting this value overrides the global columns for this version.
+	// +optional
+	AdditionalPrinterColumns []CustomResourceColumnDefinition
```

Comment:
@liggitt a detail about nil-vs-empty: setting this to `[]` will be preserved during roundtrip, right? So we can actually override with an empty list.

Reply:
not by default, no. you have to typedef the slice type and annotate it as nullable to preserve empty slices through proto round tripping. 

Because that distinction is very subtle, we try to avoid it if possible.

Reply:
Yes, avoid making this distinction meaningful. *[]Foo is what the type really should be, but that will blow up the proto converter.